documentation/src/docs/asciidoc/client/task-chain.adoc

```diff
@@ -44,40 +44,63 @@ backgroundExecutors.execute(() -> {
 As you can see this is really a lot of code for a common default workflow that you might need multiple times per application.
 
 The TaskChain API helps you to create better readable and maintainable code for scenarios like the described one.
-The TaskChain provides a fluent API that let you define workflows with multiple switches between background threads and the ui thread.
+The TaskChain provides a fluent API that let you define workflows with multiple switches between background threads and the UI thread.
 Before we have a deeper look at the API let's see how the given example would look like by using the TaskChain API:
 
 [source,java]
 ----
-UiExecutor.createUiTaskChain()
-    .execute(() -> showLoadingAnimation())
-    .background()
-    .supply(() -> callServer())
-    .ui()
-    .consume(v -> updateUi(v))
-    .onException(e -> showErrorDialog(e))
-    .thenFinally(() -> hideLoadingAnimation())
-    .run();
+UiExecutor.createUiTaskChain() // <1>
+    .execute(() -> showLoadingAnimation()) <2>
+    .background() // <3>
+    .supply(() -> callServer()) <4>
+    .map(v -> v.getName()) // <5>
+    .ui() // <6>
+    .consume(v -> updateUi(v)) // <7>
+    .onException(e -> showErrorDialog(e)) // <8>
+    .thenFinally(() -> hideLoadingAnimation()) // <9>
+    .run(); // <10>
 ----
 
-////
-todo: some description of the code
-////
+<1> a UiTaskChain is will execute the tasks in the UI thread. There is also a BackgroundTaskChain.
```

Comment:
UiTaskChain and BackgroundTaskChain names look like real classes. maybe "ui task chain" and "background task chain" to have it look more like names / descriptions instead of class names.